catalog/core/catalog-core-directorymonitor/src/test/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitorTest.java

```diff
@@ -22,322 +22,196 @@
         RouteDefinition routeDefinition = camelContext.getRouteDefinitions()
                 .get(0);
-        assertThat(routeDefinition.toString(), containsString(
-                "SetHeader[" + Constants.ATTRIBUTE_OVERRIDES_KEY + ", simple{Simple: test1=someParameter1,test2=someParameter2}"));
+        assertThat(routeDefinition.toString(),
+                containsString("SetHeader[" + Constants.ATTRIBUTE_OVERRIDES_KEY
+                        + ", simple{Simple: test1=someParameter1,test2=someParameter2}"));
     }
 
-    /**
-     * ********************************************************************************
-     */
-
-    private RouteDefinition createRouteWithAdvice(String monitoredDirectory,
-            boolean copyIngestedFiles) throws Exception {
-        camelContext = (ModelCamelContext) super.createCamelContext();
-        camelContext.start();
+    @Test
+    public void testRouteCreationMissingMonitoredDirectory() throws Exception {
+        ContentDirectoryMonitor monitor = createContentDirectoryMonitor();
+        submitConfigOptions(monitor, "", true);
+        assertThat(camelContext.getRouteDefinitions()
+                .size(), is(0));
+        assertTrue(monitor.getRouteDefinitions()
+                .isEmpty());
+    }
 
-        contentDirectoryMonitor = new ContentDirectoryMonitor(camelContext);
-        contentDirectoryMonitor.systemSubjectBinder = noOpProcessor;
-        contentDirectoryMonitor.setMonitoredDirectoryPath(monitoredDirectory);
-        contentDirectoryMonitor.setCopyIngestedFiles(copyIngestedFiles);
-
-        // Simulates what container would do once all setters have been invoked
-        contentDirectoryMonitor.init();
-
-        // Did not work because it expects the route to be "adviced" already exists. So the above
-        // init()
-        // call created the initial route with the "content:framework" node in it, and then this
-        // AdviceWithRouteBuilder replaced the content:framework with mock:result, but created a
-        // second
-        // route that had this change in it. (So still get NPE because first route fires and cannot
-        // resolve the "content" scheme)
-        camelContext.getRouteDefinitions()
-                .get(0)
-                .adviceWith(camelContext, new AdviceWithRouteBuilder() {
-                    @Override
-                    public void configure() throws Exception {
-                        // weave the node in the route which has id = content://framework
-                        // and replace it with the following route path
-                        weaveByToString(".*content:framework.*").replace()
-                                .to("mock:result");
+    private void doAndVerifyFileMove(File destinationFolder, File monitoredFolder,
+            String inputFileName) throws Exception {
+        doFileMove(destinationFolder, inputFileName);
+        Failsafe.with(new RetryPolicy().withMaxRetries(10)
```

Comment:
How can we support or handle such odd corner cases with regard to the underlying system? If the disk IO is not finishing in 10 seconds then there are much bigger problems on the user's hands. 

This seems like a problem without a real solution. If, while answering the question: 
`When does the test fail?`
we also have to answer the question: 
`Is the user's machine insufficient?`
we will be stuck in a paradox because maybe we need to wait forever.